### ISSUE TITLE
should timeout if service is down

### ISSUE THREAD

**Issue by collingreen** [StackHut/client-python#2]
it would be nice if the connection had a timeout instead of running forever

**Comment by mands** [comment]
Yep - there's a built-in timeout of 5 minutes I believe. Do you think that's enough or should it be shorter? Could even let it be defined by the service creator.

**Comment by collingreen** [comment]
Hmmmm. In this case, the client never gets info from the service, so it
might be a moot point. Still, there are two pretty different use cases -
mostly you want endpoints that respond very quickly, so 5 min is
disturbingly long. However, stackhut is cool for handling heavy, long
running tasks, so in that scenario the long timeout might be a good thing.
It might be best to just be configurable from the user side as an extra
parameter.

**Comment by mands** [comment]
Yep, that's the trade-off, for short tasks you'd want an short timeout, for longer-running computations longer timeouts make more sense (at least until we get async calls working).

Yep, I think user-side configurable timeout doable for now, could eventually let service creators define the timeout at a function level.
